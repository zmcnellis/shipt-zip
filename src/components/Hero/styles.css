/* Hero ----------------------------------------------------------------------*/
.hero {
  height: 422px;
  position: relative;
  color: white;
  width: 100%;
  box-sizing: border-box;
  text-align: center;
}

.hero-image {
  width: 100%;
  height: 100%;
  background: url('../../images/target.svg') center #8D0305 no-repeat;
  background-size: cover;
  color: white;
  text-shadow: 0px 0px 5px #000;
}

.hero-content {
  position: absolute;
  height: 100%;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.hero-container {
  position: relative;
  top: 50%;
  transform: translateY(-50%);
}

.hero-content.subtitle {
  padding: 10px 0 10px 0;
}

/* Media queries -------------------------------------------------------------*/
@media (min-width: 700px) {
  /* hero */
  .hero {
    margin: 0 auto;
    max-width: 1080px;
    padding: 12px 24px 12px 24px;
  }

  .hero .small-text {
    padding: 0 20%;
  }
}
