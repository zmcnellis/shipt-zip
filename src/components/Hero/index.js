import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchBar from '../SearchBar';
import './styles.css';

class Hero extends Component {

  render() {
    const title = "Target exclusive offer.";
    const subtitle = "$49 membership (reg. $99) + $15 credit with qualifying purchase.*";
    const detail = "*Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.";

    return (
      <section className="hero">
        <div className="hero-image"></div>
        <div className="hero-content">
          <div className="hero-container">
            <p className="title">{title}</p>
            <p className="subtitle">{subtitle}</p>

            <SearchBar handleSubmit={this.props.handleSubmit} hasError={this.props.hasError} setSearchReference={this.props.setSearchReference}/>

            <p className="small-text">{detail}</p>
          </div>
        </div>
      </section>
    );
  }
}

Hero.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  hasError: PropTypes.bool.isRequired,
  setSearchReference: PropTypes.func.isRequired
};

export default Hero;
