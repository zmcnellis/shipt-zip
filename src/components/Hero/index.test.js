import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Hero from '.';

describe('Hero', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Hero hasError={false} handleSubmit={() => "mock"}/>
    );
    expect(output).toMatchSnapshot();
  });

  it('should have search bar component', () => {
    const output = shallow(
      <Hero hasError={false} handleSubmit={() => "mock"}/>
    );
    expect(output.find('SearchBar').exists()).toBeTruthy();
  });
});
