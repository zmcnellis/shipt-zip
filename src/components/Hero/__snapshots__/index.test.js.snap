// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero
    handleSubmit={[Function]}
    hasError={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="hero-image"
/>,
        <div
          className="hero-content"
>
          <div
                    className="hero-container"
          >
                    <p
                              className="title"
                    >
                              Target exclusive offer.
                    </p>
                    <p
                              className="subtitle"
                    >
                              $49 membership (reg. $99) + $15 credit with qualifying purchase.*
                    </p>
                    <SearchBar
                              handleSubmit={[Function]}
                              hasError={false}
                    />
                    <p
                              className="small-text"
                    >
                              *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
                    </p>
          </div>
</div>,
      ],
      "className": "hero",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "hero-image",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="hero-container"
>
            <p
                        className="title"
            >
                        Target exclusive offer.
            </p>
            <p
                        className="subtitle"
            >
                        $49 membership (reg. $99) + $15 credit with qualifying purchase.*
            </p>
            <SearchBar
                        handleSubmit={[Function]}
                        hasError={false}
            />
            <p
                        className="small-text"
            >
                        *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
            </p>
</div>,
          "className": "hero-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="title"
>
                Target exclusive offer.
</p>,
              <p
                className="subtitle"
>
                $49 membership (reg. $99) + $15 credit with qualifying purchase.*
</p>,
              <SearchBar
                handleSubmit={[Function]}
                hasError={false}
/>,
              <p
                className="small-text"
>
                *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
</p>,
            ],
            "className": "hero-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Target exclusive offer.",
                "className": "title",
              },
              "ref": null,
              "rendered": "Target exclusive offer.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$49 membership (reg. $99) + $15 credit with qualifying purchase.*",
                "className": "subtitle",
              },
              "ref": null,
              "rendered": "$49 membership (reg. $99) + $15 credit with qualifying purchase.*",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleSubmit": [Function],
                "hasError": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "*Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.",
                "className": "small-text",
              },
              "ref": null,
              "rendered": "*Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="hero-image"
/>,
          <div
            className="hero-content"
>
            <div
                        className="hero-container"
            >
                        <p
                                    className="title"
                        >
                                    Target exclusive offer.
                        </p>
                        <p
                                    className="subtitle"
                        >
                                    $49 membership (reg. $99) + $15 credit with qualifying purchase.*
                        </p>
                        <SearchBar
                                    handleSubmit={[Function]}
                                    hasError={false}
                        />
                        <p
                                    className="small-text"
                        >
                                    *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
                        </p>
            </div>
</div>,
        ],
        "className": "hero",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "hero-image",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="hero-container"
>
              <p
                            className="title"
              >
                            Target exclusive offer.
              </p>
              <p
                            className="subtitle"
              >
                            $49 membership (reg. $99) + $15 credit with qualifying purchase.*
              </p>
              <SearchBar
                            handleSubmit={[Function]}
                            hasError={false}
              />
              <p
                            className="small-text"
              >
                            *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
              </p>
</div>,
            "className": "hero-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="title"
>
                  Target exclusive offer.
</p>,
                <p
                  className="subtitle"
>
                  $49 membership (reg. $99) + $15 credit with qualifying purchase.*
</p>,
                <SearchBar
                  handleSubmit={[Function]}
                  hasError={false}
/>,
                <p
                  className="small-text"
>
                  *Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.
</p>,
              ],
              "className": "hero-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Target exclusive offer.",
                  "className": "title",
                },
                "ref": null,
                "rendered": "Target exclusive offer.",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$49 membership (reg. $99) + $15 credit with qualifying purchase.*",
                  "className": "subtitle",
                },
                "ref": null,
                "rendered": "$49 membership (reg. $99) + $15 credit with qualifying purchase.*",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "handleSubmit": [Function],
                  "hasError": false,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.",
                  "className": "small-text",
                },
                "ref": null,
                "rendered": "*Membership offer valid for new members only. Target order of $100 or more must be placed to qualify for $15 credit, which expires 30 days after qualifying order is placed and can be applied to any order from Shipt.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
